config/src/utils/metricUtils.lua

```diff
@@ -0,0 +1,186 @@
+-- Copyright 2018 Yahoo Inc.
+-- Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+
+local utils = require("utils.utils")
+
+--- a module provides util functions for metrics and makers config.
+-- @module metricUtils
+
+local M = {
+    cache_makers = {}
+}
+
+-------------------------------------------------------------------------------
+-- Makers
+-------------------------------------------------------------------------------
+
+--- Parse a group of config makers and add them into a table.
+--
+-- @param makers  A group of makers
+-- @param t  The table for storing makers
+function M.add_makers(makers, t)
```

Comment:
It looks like we support either a table or a list.

Frankly, I think we should get rid of the list approach completely. First off,
we don't want to give people a bunch of different shapes our data could take.
That just makes it harder for them to learn our configuration language, and it
makes it harder for them to do their own operations on their configuration.

Our data structures should take a single shape as much as possible, unless that
single shape inhibits clarity too much, and should be as simple as possible
(though no simpler :D).

Second, I just don't see any advantages that the list approach has over the 
table approach, except for brevity. But that brevity is obtained by sacrificing
clarity, so it's not worth it.

Our goal with this configuration language should be that it's dead simple to 
use, read, and write Lua code to manipulate.